notes/zrq/20210728-01-live-deploy.txt

```diff
@@ -0,0 +1,1288 @@
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        atolmis/ansible-client:2020.12.02 \
+        bash
+
+
+# -----------------------------------------------------
+# Set the target cloud.
+#[root@ansibler]
+
+    cloudname=gaia-prod
+
+
+# -----------------------------------------------------
+# Delete everything.
+#[root@ansibler]
+
+    time \
+        /deployments/openstack/bin/delete-all.sh \
+            "${cloudname:?}"
+
+    >   real    3m15.988s
+    >   user    1m11.065s
+    >   sys     0m9.757s
+
+
+# -----------------------------------------------------
+# Create everything, using a standard config.
+#[root@ansibler]
+
+    time \
+        /deployments/hadoop-yarn/bin/create-all.sh \
+            "${cloudname:?}" \
+            'cclake-medium-04'
+
+    >   real    72m46.265s
+    >   user    16m57.612s
+    >   sys     4m48.170s
+
+
+# -----------------------------------------------------
+# Check the deployment status.
+#[root@ansibler]
+
+    cat '/tmp/aglais-status.yml'
+
+    >   aglais:
+    >     status:
+    >       deployment:
+    >         type: hadoop-yarn
+    >         conf: cclake-medium-04
+    >         name: gaia-prod-20210728
+    >         date: 20210728T231525
+    >     spec:
+    >       openstack:
+    >         cloud: gaia-prod
+
+
+# -----------------------------------------------------
+# Add the Zeppelin user accounts.
+#[root@ansibler]
+
+    ssh zeppelin
+
+        pushd "${HOME}"
+        ln -s "zeppelin-0.8.2-bin-all" "zeppelin"
+
+            pushd "zeppelin"
+
+                # Manual edit to add names and passwords
+                vi conf/shiro.ini
+
+                # Restart Zeppelin for the changes to take.
+                bin/zeppelin-daemon.sh restart
+
+            popd
+        popd
+    exit
+
+    >   Zeppelin stop                                              [  OK  ]
+    >   Zeppelin start                                             [  OK  ]
+
+
+# -----------------------------------------------------
+# Add the notebooks from github.
+#[root@ansibler]
+
+    ssh zeppelin
+
+        pushd /home/fedora/zeppelin
+
+            mv -b notebook \
+               notebook-origin
+
+	        git clone https://github.com/wfau/aglais-notebooks.git notebook
+
+	        bin/zeppelin-daemon.sh restart
+
+        popd
+    exit
+
+    >   Cloning into 'notebook'...
+    >   remote: Enumerating objects: 361, done.
+    >   remote: Counting objects: 100% (361/361), done.
+    >   remote: Compressing objects: 100% (155/155), done.
+    >   remote: Total 361 (delta 131), reused 312 (delta 85), pack-reused 0
+    >   Receiving objects: 100% (361/361), 8.78 MiB | 10.99 MiB/s, done.
+    >   Resolving deltas: 100% (131/131), done.
+
+    >   Zeppelin stop                                              [  OK  ]
+    >   Zeppelin start                                             [  OK  ]
+
```

Comment:
It looks like the notebooks are clones without the user & pass and the "post-commit" script that allows Zeppelin to commit to github, so if Nigel and/or Dennis try to commit I assume it won't work?